File: Leetcode/701-Insert-Into-A-Binary-Search-Tree/InsertIntoABinarySearchTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def insertIntoBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        node = TreeNode(val)
        if not root:
            root = node
            return root
        cur = root
        while True:
            if cur.val > val:
                if not cur.left:
                    cur.left = node
                    break
                cur = cur.left
                continue
            if not cur.right:
                cur.right = node
                break
            cur = cur.right
        return root

File: Leetcode/701-Insert-Into-A-Binary-Search-Tree/InsertIntoABinarySearchTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        TreeNode* node = new TreeNode(val);
        if (root == NULL) {
            root = node;
            return root;
        }
        TreeNode* cur = root;
        while(true) {
            if (cur->val > val) {
                if (cur->left == NULL) {
                    cur->left = node;
                    break;
                }
                cur = cur->left;
                continue;
            }
            if (cur->right == NULL) {
                cur->right = node;
                break;
            }
            cur = cur->right;
        }
        return root;
    }
};
